### DISCUSSION TITLE
Setting up ESP8266 in soft access point mode

### DISCUSSION THREAD

**Discussion by svanimisetti** [lathoub/Arduino-AppleMIDI-Library#117] [Q&A] [answered]
After much struggle, I was able to get the AppleMIDI library example (_ESP8266_NoteOnOffEverySec.ino_) running on my ESP8266. I notice that when my wi-fi network is congested, I see lot of dropped notes. I wanted to know how I can setup ESP8266 in soft access point mode so that my iPad can connect to the ESP8266 directly. I am assuming that the performance will be better in this case, as there will be less traffic. Any comments or suggestions are greatly appreciated. Thanks in advance.

**Comment by svanimisetti** [comment, marked as answer]
Sorry - had double posted this to [FortySevenEffect discussion](https://github.com/FortySevenEffects/arduino_midi_library/discussions/224) by mistake.

Answering my own question. I found that simple modification to existing example actually works (found it by trial and error).

On line 25, use WiFi.softAP instead of WiFi.begin
Comment out lines 27-31
On line 34, use WiFi.softAPIP() instead of WiFi.localIP()
With above changes, I am able to connect to ESP8266 over WiFi and recieve a note every second on test application (Synthesia). Tested using rtpMIDI (Tobia Erichsen) on Windows 10 and MIDI Network (Oscar Bouwmans) on iOS 9.x (very old iPad).

**Reply by lathoub** [reply]
Thank for the research @svanimisetti , If you provide a PR with a minimal example, I'll add it to the repo
